File: backend/app/models/candidate.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, Enum, ForeignKey
from sqlalchemy.orm import relationship

from app.database import Base


class Candidate(Base):
    __tablename__ = "candidates"

    candidate_id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(255), nullable=False)
    email = Column(String(255), unique=True)
    phone = Column(String(50))
    location = Column(String(255))
    gdrive_file_id = Column(String(255))
    cv_s3_path = Column(String(500))
    raw_cv_text = Column(Text)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    skills = relationship(
        "CandidateSkill",
        back_populates="candidate",
        cascade="all, delete-orphan"
    )
    experiences = relationship(
        "CandidateExperience",
        back_populates="candidate",
        cascade="all, delete-orphan"
    )
    evaluations = relationship("Evaluation", back_populates="candidate")
    contacts = relationship("Contact", back_populates="candidate")
    statuses = relationship("CandidateStatus", back_populates="candidate")
    feedbacks = relationship("InterviewFeedback", back_populates="candidate")


class CandidateSkill(Base):
    __tablename__ = "candidate_skills"

    skill_id = Column(Integer, primary_key=True, autoincrement=True)
    candidate_id = Column(Integer, ForeignKey("candidates.candidate_id"))
    skill_name = Column(String(255), nullable=False)
    skill_type = Column(Enum("technical", "soft"), default="technical")
    proficiency_level = Column(String(50))

    candidate = relationship("Candidate", back_populates="skills")


class CandidateExperience(Base):
    __tablename__ = "candidate_experience"

    experience_id = Column(Integer, primary_key=True, autoincrement=True)
    candidate_id = Column(Integer, ForeignKey("candidates.candidate_id"))
    company = Column(String(255))
    position = Column(String(255))
    duration_months = Column(Integer)
    start_date = Column(String(50))
    end_date = Column(String(50))
    description = Column(Text)

    candidate = relationship("Candidate", back_populates="experiences")

File: backend/HireGenius.API/Models/Candidate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HireGenius.API.Models;

/// <summary>
/// Represents a job candidate in the system.
/// </summary>
[Table("candidates")]
public class Candidate
{
    [Key]
    [Column("candidate_id")]
    public Guid CandidateId { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(255)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(255)]
    [Column("email")]
    public string Email { get; set; } = string.Empty;

    [MaxLength(50)]
    [Column("phone")]
    public string? Phone { get; set; }

    [MaxLength(255)]
    [Column("location")]
    public string? Location { get; set; }

    [MaxLength(255)]
    [Column("gdrive_file_id")]
    public string? GDriveFileId { get; set; }

    [MaxLength(500)]
    [Column("cv_s3_path")]
    public string? CvS3Path { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public ICollection<CandidateSkill> Skills { get; set; } = new List<CandidateSkill>();
    public ICollection<CandidateExperience> Experiences { get; set; } = new List<CandidateExperience>();
    public ICollection<Evaluation> Evaluations { get; set; } = new List<Evaluation>();
    public ICollection<Contact> Contacts { get; set; } = new List<Contact>();
    public ICollection<InterviewFeedback> InterviewFeedbacks { get; set; } = new List<InterviewFeedback>();
    public ICollection<CandidateStatus> Statuses { get; set; } = new List<CandidateStatus>();
}
